item.go

```diff
@@ -62,6 +71,7 @@ func (item *Item[K, V]) update(value V, ttl time.Duration) {
 	defer item.mu.Unlock()
 
 	item.value = value
+	item.cost = item.calculateCost(item)
```

Comment:
The cost needs to be set in `NewItemWithOpts` as well. `update()` isn't used there.

Reply:
By default, if the `WithMaxCost` function is not used, the cost is always `0` and the default implementation of `calculateCost` just returns `0`. You can see that in the `newItemWithOpts` implementation.

If the `WithMaxCost` option is used, it will result in a call to `withCostFunc`, which will set the `costFunc` on the item and also call it. I'll move this call to the implementation of the `newItemWithOpts` function.

Reply:
https://github.com/jellydator/ttlcache/pull/152/commits/5cb43bf6bac41c894f108681075e00dcc1f22484 make it more clear

Reply:
Ah yes, I missed that part. I think your updated approach is clearer indeed :+1: